src/fieldAnalyser/fieldBuilder.ts

```diff
@@ -1,24 +1,19 @@
 import {FieldModel, FieldTypes} from '@coveord/platform-client';
 
-export class FieldBuilder {
-  public readonly dict: Record<string, FieldTypes> = {};
-
-  constructor() {}
-  public add(name: string, type: FieldTypes) {
-    this.dict[name] = type;
-    return this;
+export class FieldBuilder extends Map<string, FieldTypes> {
```

Comment:
Mh the change is not that inconsequential here. Is a FieldBuilder really a map?
In that case maybe it should have a name that explain that it's just a data holding structure.

Reply:
Yes, it's really just a map holding data. I agree, the name is misleading.